WordPress/gutenberg
# Revert: Make sticky block action on template part block

## What?


Revert #49085, which added in a "Make sticky" action in the block settings for the Template Part block (available within the list view and the block toolbar)

## Why?


#49085 was a slightly contentious feature! 😄

While the feature intended to make it easier for a user to create a sticky header, it did so by wrapping a template part block in a Group block, and then sets that Group block to sticky. The feedback on the feature is that while this is currently the only way a user can create a sticky header within the site editor, it isn't a flow that we want to optimise for, and ideally there would be a more intuitive way for users to achieve this. See: https://github.com/WordPress/gutenberg/pull/49085#issuecomment-1477276305 for more discussion.

By reverting, we can give a bit more time for the design discussion to evolve, and we can try out other ideas in subsequent PRs. Let's keep the discussion going! Perhaps over in https://github.com/WordPress/gutenberg/issues/47043.

Thanks for all the reviews, discussion, and feedback, everybody!

## How?


Revert the changes in #49085.

## Testing Instructions





Smoke test by opening up the site editor, and selecting a Header template part at the root of the document. In the block settings menu within both the list view and the block toolbar, you should _not_ see "Make sticky" as an option, and you _should_ see "Detach blocks from template part".

Double check that the removals all look okay — the added exports in `private-apis.js` were only added in the earlier PR and aren't being used anywhere else as far as I can see, so I've reverted them, too.

## Screenshots or screencast 

<img width="980" alt="image" src="https://user-images.githubusercontent.com/14988353/226524595-60499cee-c8a4-461c-aa30-8e84dfce8853.png">
